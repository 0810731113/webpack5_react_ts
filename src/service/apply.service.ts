import { AddInterceptors } from 'function/interceptors';
import { EnterpriseApi, PersonalAccountApi } from "api-auth";
import {} from "api/generated/model";
import Axios, { AxiosStatic, AxiosInstance } from "axios";

import {
  EnterpriseAccountCreation,
  PersonalAccountCreation,
} from "api-auth/generated/model";

export class ApplyService {
  enterpriseApi: EnterpriseApi;

  personalAccountApi: PersonalAccountApi;

  constructor(baseUrl: string, axios: AxiosInstance) {
    this.enterpriseApi = new EnterpriseApi({}, baseUrl, axios);
    this.personalAccountApi = new PersonalAccountApi({}, baseUrl, axios);


    AddInterceptors(axios, "apply");
  }

  async sendSMS(mobile: string) {
    const response = await this.enterpriseApi.getCodeUsingGET(mobile, "query");
    return response.data;
  }

  async getMainAccountList(mobile: string, code: string) {
    const response = await this.enterpriseApi.getAccountListUsingGET(
      mobile,
      code,
    );
    return response.data.data;
  }

  async verifyUserPassword(userName: string, userPassword: string) {
    const response = await this.enterpriseApi.getUserInfoUsingPOST({
      userName,
      userPassword,
    });
    return response.data.data;
  }

  async submitApply(isNew: boolean, data: EnterpriseAccountCreation) {
    // if (isNew) {
    //   const response = await this.enterpriseApi.createEnterpriseAccountUsingPOST(
    //     data,
    //   );
    //   return response.data.data;
    // } else {
    const response = await this.enterpriseApi.reviewEnterpriseAccountUsingPOST(
      data,
    );
    return response.data.data;
  }

  async registerUserCenter(
    identity: string,
    password: string,
    passwordMobile: string,
  ) {
    const response = await this.enterpriseApi.createEnterpriseAccountV2UsingPOST(
      { identity, password, passwordMobile },
    );
    return response.data.data;
  }

  async reviewPersonalAccount(data: PersonalAccountCreation) {
    const response = await this.personalAccountApi.reviewPersonalAccountUsingPOST(
      data,
    );
    return response.data.data;
  }

  async getCode(mobile: string) {
    console.log(mobile);
    const response = await this.personalAccountApi.getCodeUsingGET1(
      mobile,
      "query",
    );
    return response.data;
  }
}
