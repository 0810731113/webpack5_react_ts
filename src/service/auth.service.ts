import { notification } from "antd";
import { isPanel } from "consts";
import { getLoginUrl } from "function/auth.func";
import { getCookie, setCookie } from "function/cookie.func";

export class AuthService {
  getToken(handleLogin = true) {
    const token = getCookie("accessToken");
    if (handleLogin && !token) {
      this.onError();
    }
    return token;
  }

  getUserId(handleLogin = true) {
    const userId = getCookie("userId");
    // return userId ?? "6399205335958287204";
    if (handleLogin && !userId) {
      this.onError();
    }
    return userId || "";
  }

  onError() {
    window.location.href = getLoginUrl();
    throw new Error("请重新登录");
  }
}

export interface RequestErrorHandler {
  onServerError(message?: string): void;

  onNotFound(message?: string): void;

  onUnauthorized(message?: string): void;

  onForbidden(message?: string): void;

  onInvalidParameters(message?: string): void;

  onUnknown(message?: string): void;
}

class WebRequestErrorHandler implements RequestErrorHandler {
  onServerError(message?: string): void {
    notification.error({
      message: "服务器内部错误",
      description: message ?? "请稍后重试",
    });
  }

  onNotFound(message?: string): void {
    notification.error({
      message: "接口不存在",
      description: message ?? "请检查 URL 和 HTTP Method",
    });
  }

  onUnauthorized(message?: string): void {
    notification.error({
      message: "用户未登录",
      description: message ?? "请重新登录",
    });
    setTimeout(() => {
      window.location.href = getLoginUrl();
    }, 2000);
  }

  onInvalidParameters(message?: string): void {
    notification.error({
      message: "参数错误",
      description: message ?? "请检查参数",
    });
  }

  onForbidden(message?: string): void {
    notification.error({
      message: "无权限访问",
      description: message ?? "请检查用户角色",
    });
  }

  onUnknown(message?: string): void {
    notification.error({
      message: "未知错误",
      description: `${message} 请查看控制台获取错误详情`,
    });
  }
}

class PanelRequestErrorHandler extends WebRequestErrorHandler {
  onUnauthorized() {
    notification.error({
      message: "用户未登录",
    });
    setCookie("userId", undefined);
    setCookie("accessToken", undefined);
  }
}

const requestErrorHandler: RequestErrorHandler = isPanel
  ? new PanelRequestErrorHandler()
  : new WebRequestErrorHandler();

export { requestErrorHandler };
