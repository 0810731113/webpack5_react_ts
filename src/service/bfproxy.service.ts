import { BimfaceControllerApi, TranslateControllerApi } from "api-bfproxy";
import {} from "api/generated/model";

import Axios, { AxiosStatic } from "axios";

export class BfproxyService {

  bimfaceApi: BimfaceControllerApi;

  translateApi: TranslateControllerApi;

  constructor(baseUrl: string, axios: AxiosStatic) {
    this.bimfaceApi = new BimfaceControllerApi({}, baseUrl, axios);
    this.translateApi = new TranslateControllerApi({}, baseUrl, axios);
  }

  async getFileDownloadUrl(fileId: string) {
    const response = await this.bimfaceApi.getFileDownloadUrlUsingGET(fileId);
    return response.data as any;
  }

  async getViewToken(fileId: string) {
    const response = await this.bimfaceApi.getViewTokenUsingGET(fileId);
    return response.data
  }

  async startWorker(versionId: number, type: string) {
    const data = {
      versionId,
      toType: type,
    };

    const response = await this.translateApi.addJobUsingPOST(data);
    return response.data;
  }
}
