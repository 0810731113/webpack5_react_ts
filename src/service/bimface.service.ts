import { first, isArray } from "lodash";
import consts from "consts";

declare const BimfaceSDKLoaderConfig: any;
declare const BimfaceSDKLoader: any;
declare const Glodon: any;
export interface BimfaceServiceConfig {
  domId: string;
  format: string;
  viewTokens: string | string[];
  toolbars?: string[];
  cameraAnimation?: boolean;
}
export class BimfaceService {
  initViewToken: string | null = null;

  domId: string;

  format: string;

  viewTokens: string[];

  toolbars: string[] = ["MainToolbar"];

  cameraAnimation?: boolean = false;

  _app: any;

  constructor(
    config: BimfaceServiceConfig,
    private onViewerCreated: (app: any, viewer: any) => void,
    private onLoaded?: () => void,
    private onViewAdded?: (viewToken: string) => void,
  ) {
    this.domId = config.domId;
    this.format = config.format;
    this.viewTokens = isArray(config.viewTokens)
      ? config.viewTokens
      : [config.viewTokens];
    this.cameraAnimation = config.cameraAnimation;
    this.toolbars = config.toolbars ?? ["MainToolbar"];
    this.initViewToken = first(this.viewTokens) ?? null;
  }

  start2D(viewToken: string, domElement: HTMLElement) {
    const config = new Glodon.Bimface.Application.WebApplicationDrawingConfig();
    config.domElement = domElement;
    const app = new Glodon.Bimface.Application.WebApplicationDrawing(config);
    app.load(viewToken);
    return app;
  }

  start3D(viewToken: string, domElement: HTMLElement) {
    const config = new Glodon.Bimface.Application.WebApplication3DConfig();
    config.domElement = domElement;
    config.Toolbars = this.toolbars;
    this._app = new Glodon.Bimface.Application.WebApplication3D(config);
    const view3d = this._app.getViewer();
    view3d.render();
    return this._app;
  }

  init() {
    const loaderConfig = new BimfaceSDKLoaderConfig();
    loaderConfig.viewToken = this.initViewToken;
    loaderConfig.version = "3.6.102";
    BimfaceSDKLoader.load(
      loaderConfig,
      (viewMetaData: any) => {
        if (!this.initViewToken) {
          return;
        }

        this.onLoaded && this.onLoaded();

        const domElement = document.getElementById(this.domId)!;
        const is3D = this.format !== "dwg";
        let app: any = null;
        if (is3D) {
          app = this.start3D(this.initViewToken, domElement);
        } else {
          app = this.start2D(this.initViewToken, domElement);
        }

        const viewer = app.getViewer();
        this.onViewerCreated && this.onViewerCreated(app, viewer);
        viewer.setCameraAnimation &&
          viewer.setCameraAnimation(this.cameraAnimation);
      },
      console.error,
    );
  }
}
