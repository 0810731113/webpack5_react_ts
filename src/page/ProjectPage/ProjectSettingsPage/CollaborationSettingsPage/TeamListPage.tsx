/* eslint-disable @typescript-eslint/no-use-before-define */
import { message, Popconfirm, Space, Tag, Button, Table } from "antd";
import { ColumnType } from "antd/es/table/interface";
import { Team } from "api/generated/model";
import useTeamUsers from "hook/use-team-users.hook";
import ProjectPageContext from "page/ProjectPage/ProjectPageContext";
import React, { useContext } from "react";
import { useRouteMatch } from "react-router";
import { Link } from "react-router-dom";
import { useRecoilValue } from "recoil";
import { teamService } from "service";
import projectPageState from "state/project.state";
import { useImmer } from "use-immer";
import { CheckPermission } from "component/CheckPermission/CheckPermission";
import { ResourcePermissionResourceEnum } from "api-authorization/generated/model";
import { TooltipWrapper } from "component/wrapper/TooltipWrapper";
import { PopconfirmWrapper } from "component/wrapper/PopconfirmWrapper";
import AddTeamDrawer from "./components/AddTeamDrawer";

export interface TeamListPageProps {}

export interface State {
  showAddTeamForm: boolean;
  selectedTeamId?: string | null;
}

export default function TeamListPage(props: TeamListPageProps) {
  const {} = props;
  const [{ showAddTeamForm, selectedTeamId }, updateState] = useImmer<State>({
    showAddTeamForm: false,
  });
  const {
    url,
    path,
    params: {},
  } = useRouteMatch<{}>();
  const { onTeamNotFound, onResponseError } = useContext(ProjectPageContext);
  const { teams } = useRecoilValue(projectPageState);
  const columns: ColumnType<Team>[] = [
    {
      title: "团队名称",
      dataIndex: "name",
      key: "name",
      render(row, record) {
        return <Link to={`${url}/${record.id}`}>{record.name}</Link>;
      },
    },
    {
      title: "成员",
      key: "users",
      render(row, record) {
        return <TeamMemberList teamId={record.id!} />;
      },
    },
    {
      title: "操作",
      key: "operation",
      render(row, record) {
        return (
          <Space>
            <CheckPermission
              resouseType={ResourcePermissionResourceEnum.CollaborationSetting}
            >
              <TooltipWrapper
                when={(tooltipProps) => tooltipProps.disabled ?? false}
                title="处于示例项目中无该功能权限"
              >
                <Button
                  type="link"
                  onClick={() => {
                    updateState((draft) => {
                      draft.selectedTeamId = record.id;
                      draft.showAddTeamForm = true;
                    });
                  }}
                >
                  编辑
                </Button>
              </TooltipWrapper>
            </CheckPermission>
            <CheckPermission
              resouseType={ResourcePermissionResourceEnum.CollaborationSetting}
            >
              <TooltipWrapper
                when={(tooltipProps) => tooltipProps.disabled ?? false}
                title="处于示例项目中无该功能权限"
              >
                <PopconfirmWrapper
                  title="确认删除团队吗？"
                  okButtonProps={{ danger: true, type: "primary" }}
                  onConfirm={() => {
                    teamService
                      .deleteTeam(record.id!)
                      .then(() => {
                        message.success("团队已经删除");
                        onTeamNotFound("");
                      })
                      .catch(onResponseError);
                  }}
                >
                  <Button type="link">删除</Button>
                </PopconfirmWrapper>
              </TooltipWrapper>
            </CheckPermission>
          </Space>
        );
      },
    },
  ];
  return (
    <div>
      <div className="split-toolbar">
        <CheckPermission
          resouseType={ResourcePermissionResourceEnum.CollaborationSetting}
        >
          <TooltipWrapper
            when={(tooltipProps) => tooltipProps.disabled ?? false}
            title="处于示例项目中无该功能权限"
          >
            <Button
              type="primary"
              onClick={() =>
                updateState((draft) => {
                  draft.selectedTeamId = null;
                  draft.showAddTeamForm = true;
                })
              }
            >
              创建团队
            </Button>
          </TooltipWrapper>
        </CheckPermission>
      </div>
      <Table
        columns={columns}
        dataSource={teams}
        pagination={false}
        rowKey={(record) => record.id ?? ""}
      />
      {showAddTeamForm && (
        <AddTeamDrawer
          teamId={selectedTeamId}
          editMode={!!selectedTeamId}
          visible={showAddTeamForm}
          onClose={() => {
            updateState((draft) => {
              draft.showAddTeamForm = false;
            });
          }}
          onComplete={() => {
            onTeamNotFound("");
            updateState((draft) => {
              draft.showAddTeamForm = false;
            });
          }}
        />
      )}
    </div>
  );
}

function TeamMemberList({ teamId }: { teamId: string }) {
  const { users } = useTeamUsers(teamId);
  if (!users) {
    return null;
  }
  return (
    <>
      {users.map((user) => (
        <Tag key={user.id}>{user.name}</Tag>
      ))}
    </>
  );
}
