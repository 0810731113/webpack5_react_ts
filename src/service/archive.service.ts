import { ArchiveApi, VersionApi, DataSetApi } from "api";
import { AxiosStatic } from "axios";
import {
  ArchivePackageCreation,
  ArchiveVersionVO,
  DataSetVO,
  VersionVO,
  Resource,
} from "api/generated/model";
import { authService } from "service";
import { ArchivePackageVO } from "../api/generated/model/archive-package-vo";

export interface ArchivePackageInfo {
  package: ArchivePackageVO;
  versions: ArchiveVersionVO[];
}

export interface PackageVersionInfo {
  dsInfo: DataSetVO;
  vsInfo: VersionVO;
  key: string | number;
}

export class ArchiveService {
  archiveapi: ArchiveApi;

  versionapi: VersionApi;

  datasetapi: DataSetApi;

  constructor(baseUrl: string, axios: AxiosStatic) {
    this.archiveapi = new ArchiveApi({}, baseUrl, axios);
    this.versionapi = new VersionApi({}, baseUrl, axios);
    this.datasetapi = new DataSetApi({}, baseUrl, axios);
  }

  async createArchive(projectId: string, name: string, description: string) {
    const body: ArchivePackageCreation = {
      description,
      name,
      projectId,
    };
    const res = await this.archiveapi.createArchivePackageUsingPOST(
      authService.getUserId()!,
      body,
    );
    return res.data.data;
  }

  async updateArchive(
    packageId: string,
    packageVersionId: number,
    name: string,
    description: string,
  ) {
    const body = {
      description,
      name,
      packageId,
      versionId: packageVersionId,
    };
    const res = await this.archiveapi.updateArchivePackageUsingPUT(
      authService.getUserId()!,
      body,
    );
    return res.data.data;
  }

  async deleteArchive(pkgId: string) {
    const res = await this.archiveapi.dropArchivePackageUsingDELETE(
      pkgId,
      authService.getUserId()!,
    );
    return res.data.data;
  }

  async getAllArchivesVersions(projectId: string) {
    const response = await this.archiveapi.getArchivePackagesByProjectIdUsingGET(
      projectId,
    );

    const packages = response.data.data ?? [];
    return packages;
  }

  // async batchLoadOldArchivesVersions(packageIds: string[]) {
  //   const response = await this.archiveapi.getArchiveVersionsUsingPOST(
  //     packageIds,
  //   );
  //   const pkgVersions = response.data.data ?? [];
  //   return pkgVersions.reduce<{ [pkgId: string]: ArchiveVersionVO[] }>(
  //     (pre, current) => {
  //       pre[current.packageId!] = pre[current.packageId!] || [];
  //       pre[current.packageId!].push(current);
  //       return pre;
  //     },
  //     {},
  //   );
  // }

  async batchLoadArchivesVersions(packageIds: string[]) {
    const response = await this.archiveapi.getArchiveVersions4V2UsingPOST(
      packageIds,
    );
    const pkgVersions = response.data.data ?? [];
    return pkgVersions.reduce<{ [pkgId: string]: ArchiveVersionVO[] }>(
      (pre, current) => {
        pre[current.packageId!] = pre[current.packageId!] || [];
        pre[current.packageId!].push(current);
        return pre;
      },
      {},
    );
  }

  async getArchiveByPackageId(packageId: string) {
    const response = await this.archiveapi.getArchivePackageUsingGET(packageId);
    const pkg = response.data.data;
    return pkg;
  }

  // async getOldArchiveInfoByPackageId(packageId: string) {
  //   let response = await this.archiveapi.getArchivePackageUsingGET(packageId);
  //   let pkg = response.data.data;
  //   let res = await this.archiveapi.getArchiveVersionsByPackageIdUsingGET(
  //     packageId,
  //   );
  //   let versions = res.data.data;
  //   versions?.sort((a, b) => (b.version ?? 0) - (a.version ?? 0));
  //   return { package: pkg!, versions: versions! };
  // }

  async getArchiveInfoByPackageId(packageId: string) {
    const response = await this.archiveapi.getArchivePackageUsingGET(packageId);
    const pkg = response.data.data;
    const res = await this.archiveapi.getArchiveVersionsByPackageId4V2UsingGET(
      packageId,
    );
    const versions = res.data.data;
    versions?.sort((a, b) => (b.version ?? 0) - (a.version ?? 0));
    return { package: pkg!, versions: versions! };
  }

  async postOldAchiveResources(packageId: string, body: Resource[]) {
    const response = await this.archiveapi.addResourcesToPackageUsingPOST(
      packageId,
      authService.getUserId()!,
      body,
    );

    return response.data;
  }

  async postAchiveResources(packageId: string, versionIds: number[]) {
    const body = {
      deliverUnits: versionIds,
    };
    const response = await this.archiveapi.addResourcesToPackage4v2UsingPOST(
      packageId,
      authService.getUserId()!,
      body,
    );

    return response.data;
  }

  async postDraftAchiveResources(packageId: string, versionIds: number[]) {
    const body = {
      deliverUnits: versionIds,
    };
    const response = await this.archiveapi.addResourcesToDraftPackageUsingPOST(
      packageId,
      authService.getUserId()!,
      body,
    );

    return response.data;
  }

  async updateDraftAchiveResources(
    packageId: string,
    versionIds: number[],
    pkgVersionId: number,
  ) {
    const body = {
      deliverUnits: versionIds,
    };
    const response = await this.archiveapi.updateResourcesToDraftPackageUsingPUT(
      packageId,
      pkgVersionId,
      authService.getUserId()!,
      body,
    );

    return response.data;
  }

  async commitDraftAchiveResources(
    packageId: string,
    versionIds: number[],
    pkgVersionId: number,
  ) {
    const body = {
      deliverUnits: versionIds,
    };
    const response = await this.archiveapi.saveDraftArchivePackage2CommitStatusUsingPUT(
      packageId,
      pkgVersionId,
      authService.getUserId()!,
      body,
    );

    return response.data;
  }

  async getPackageVersionIds(packageId: string, versionId: number) {
    const resp = await this.archiveapi.getArchiveVersionByPackageVersionUsingGET(
      packageId,
      versionId,
    );

    const obj = resp.data.data;

    if (obj && obj.content) {
      return obj.content
        .filter((record) => record.type === "Version")
        .map((record) => parseInt(record.resourceId!));
    } 
      return [];
    
  }

  async getPackageVersionInfo(packageId: string, versionId: number) {
    const result: PackageVersionInfo[] = [];

    if (packageId && versionId) {
      const versionIds = await this.getPackageVersionIds(packageId, versionId);
      if (versionIds && versionIds.length > 0) {
        const resVer = await this.versionapi.getVersionsByIdsAndMvdsUsingPOST(
          versionIds,
        );
        const versions = resVer.data.data!;

        const dsIds = versions.map((version) => version.dataSetId!);
        const resDs = await this.datasetapi.getDataSetByIdsUsingPOST(dsIds);
        const datasets = resDs.data.data!;
        // let teams = await teamapi.findTeamsByDsIds(dsIds);

        versions.forEach((version) => {
          // let team = teams.find(team => team.dsId === version.dataSetId);

          result.push({
            dsInfo: datasets.find((ds) => ds.id === version.dataSetId)!,
            vsInfo: version,
            // teamInfo: team ? team.teamVO : null,
            key: version.id!,
          });
        });
      }
    }

    return result;
  }

  async deliveryToRecipient(pkgId: string, userId: string, projectId: string) {
    const res = await this.archiveapi.assignArchive2RecipientsUsingPOST(
      pkgId,
      projectId,
      1,
      authService.getUserId(),
      [userId],
    );
    return res.data.data;
  }

  async getArchiveRecipients(pkgId: string, projectId: string) {
    const res = await this.archiveapi.findRecipients4ResouceUsingGET(
      pkgId,
      projectId,
      1,
      authService.getUserId()!,
    );
    return res.data.data;
  }

  async disableAssign(pkgId: string, projectId: string, recipientId: string) {
    const res = await this.archiveapi.disable4RecipientsUsingPUT(
      pkgId,
      projectId,
      1,
      authService.getUserId()!,
      [recipientId],
    );
    return res.data.data;
  }

  async enableAssign(pkgId: string, projectId: string, recipientId: string) {
    const res = await this.archiveapi.enable4RecipientsUsingPUT(
      pkgId,
      projectId,
      1,
      authService.getUserId()!,
      [recipientId],
    );
    return res.data.data;
  }
}
