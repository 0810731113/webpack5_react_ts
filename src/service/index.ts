import Axios from "axios";
import consts from "consts";
import { AddInterceptors } from "function/interceptors";
import { ApplyService } from "./apply.service";
import { ArchiveService } from "./archive.service";
import { AuthService } from "./auth.service";
import { BfproxyService } from "./bfproxy.service";
import { ClassificationService } from "./classification.service";
import { ConstructionSettingsService } from "./construction-settings.service";
import ElementFilterService from "./element-filter.service";
import ElementService from "./element.service";
import { EnterpriseService } from "./enterprise.service";
import { FileService } from "./file.service";
import { FolderService } from "./folder.service";
import { GridSettingsService } from "./grid.settings.service";
import { IntegratePackagesService } from "./integrate-package.service";
import { IssueService } from "./issue.service";
import { LoginService } from "./login.service";
import { MessageService } from "./message.service";
import { PackageService } from "./package.service";
import { PermissionService } from "./permission.service";
import { ProjectService } from "./project.service";
import { RoleService } from "./role.service";
import { SpaceSettingsService } from "./space-settings.service";
import { SpecialtyService } from "./specialty.service";
import { StructureService } from "./structure.service";
import { SystemService } from "./system.service";
import { TeamService } from "./team.service";
import { TemplateService } from "./template.service";
import { UserService } from "./user.service";
import { VersionService } from "./version.service";
import { WorkUnitService } from "./work-unit.service";
import { XmonitorService } from "./xmonitor.service";
import { FontService } from "./font.service";

const {
  API_BASE_STRUC_URL,
  API_BASE_URL,
  API_BASE_BIMCODE_URL,
  API_BASE_BIMFACE_URL,
  API_BASE_SYSTEM_URL,
  API_BASE_STORE_URL,
  API_BASE_AUTHORIZATION_URL,
  AUTH_BASE_URL,
} = consts;

const authService = new AuthService();
const projectService = new ProjectService(API_BASE_URL, Axios);
const teamService = new TeamService(API_BASE_URL, Axios);
const folderService = new FolderService(API_BASE_URL, Axios);
const workUnitService = new WorkUnitService(API_BASE_URL, Axios);
const xmonitorService = new XmonitorService(API_BASE_URL, Axios);
const userService = new UserService(API_BASE_URL, Axios);
const packageService = new PackageService(API_BASE_URL, Axios);
const fileService = new FileService(API_BASE_URL, Axios);
const spaceSettingsService = new SpaceSettingsService(API_BASE_URL, Axios);
const gridSettingsService = new GridSettingsService(API_BASE_URL, Axios);
const messageService = new MessageService(API_BASE_URL, Axios);
const specialtyService = new SpecialtyService(API_BASE_URL, Axios);
const templateService = new TemplateService(API_BASE_URL, Axios);
const constructionSettingsService = new ConstructionSettingsService(
  API_BASE_STORE_URL,
  Axios,
);
const structureService = new StructureService(API_BASE_STRUC_URL, Axios);
const fontService = new FontService(API_BASE_STRUC_URL, Axios);
const archiveService = new ArchiveService(API_BASE_URL, Axios);
const integratePackagesService = new IntegratePackagesService(
  API_BASE_URL,
  Axios,
);
const permissionService = new PermissionService(API_BASE_URL, Axios);
const issueService = new IssueService(API_BASE_URL, Axios);
const versionService = new VersionService(API_BASE_URL, Axios.create());
const elementService = new ElementService(API_BASE_URL, Axios);
const classificationService = new ClassificationService(
  API_BASE_BIMCODE_URL,
  Axios,
);
const bfproxyService = new BfproxyService(API_BASE_BIMFACE_URL, Axios);
const systemService = new SystemService(API_BASE_SYSTEM_URL, Axios.create());
const applyService = new ApplyService(AUTH_BASE_URL, Axios.create());
const enterpriseService = new EnterpriseService(AUTH_BASE_URL, Axios.create());
const loginService = new LoginService(AUTH_BASE_URL, Axios.create());
const elementFilterService = new ElementFilterService(API_BASE_URL, Axios);
const roleService = new RoleService(API_BASE_AUTHORIZATION_URL, Axios);

AddInterceptors(Axios);

export {
  authService,
  fontService,
  projectService,
  teamService,
  folderService,
  workUnitService,
  userService,
  packageService,
  fileService,
  spaceSettingsService,
  gridSettingsService,
  messageService,
  specialtyService,
  templateService,
  constructionSettingsService,
  structureService,
  archiveService,
  integratePackagesService,
  permissionService,
  issueService,
  versionService,
  classificationService,
  bfproxyService,
  elementService,
  systemService,
  loginService,
  elementFilterService,
  roleService,
  xmonitorService,
  applyService,
  enterpriseService,
};
