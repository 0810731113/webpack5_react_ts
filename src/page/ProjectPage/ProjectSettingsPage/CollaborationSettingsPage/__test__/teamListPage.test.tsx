import React, { PropsWithChildren } from "react";
import pretty from "pretty";
import { render, screen } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import { render as renderHtml, unmountComponentAtNode } from "react-dom";
import { RecoilRoot, useRecoilState, atom } from "recoil";
import projectPageState from "state/project.state";
import { teamService } from "service";

import TeamListPage from "../TeamListPage";

let container: HTMLDivElement;

const defaultState = {
  projectId: "",
  project: null,
  folders: [],
  teams: [
    {
      creationTime: "2020-12-16T06:51:05.000+0000",
      id: "f058f8e7-b946-4e3f-8a8a-956146f881b8",
      name: "机电团队",
      ownerId: "6742328225488007946",
      projectId: "fad4e43c-409f-4c0d-aaa9-ad4366a078d8",
    },
    {
      creationTime: "2020-12-16T06:50:05.000+0000",
      id: "6d52f9de-2c67-476c-94f3-c143b83d6ccf",
      name: "结构团队",
      ownerId: "6742328225488007946",
      projectId: "fad4e43c-409f-4c0d-aaa9-ad4366a078d8",
    },
    {
      creationTime: "2020-12-16T06:49:55.000+0000",
      id: "c98c4230-f2c9-4af6-9e5e-7abcd2c39141",
      name: "建筑团队",
      ownerId: "6742328225488007946",
      projectId: "fad4e43c-409f-4c0d-aaa9-ad4366a078d8",
    },
  ],
  myTeams: null,
  specialties: [],
  users: [],
  currentUser: {
    userName: "诸大大的工作室",
    name: "诸大大的工作室",
    id: "6742328225488007946",
  },
  templates: [],
  roles: [],
  resources: [],
};

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  if (container) {
    unmountComponentAtNode(container);
    container.remove();
  }
  jest.clearAllMocks();
});

jest.mock("react-router", () => ({
  useRouteMatch: () => ({
    url: "/",
    params: { projectId: "xxx" },
  }),
}));

jest.mock("react-router-dom", () => ({
  Link: (props: PropsWithChildren<{ to: string }>) => (
    <a href={props.to}>{props.children}</a>
  ),
  useHistory: () => ({
    push: jest.fn(),
    replace: jest.fn(),
  }),
}));

jest.mock("service");

test("test team list page", async () => {
  renderHtml(
    <RecoilRoot
      initializeState={(snap) => snap.set(projectPageState, defaultState)}
    >
      <TeamListPage />
    </RecoilRoot>,
    container,
  );

  await act(async () => {
    await (teamService.getUsersInTeams);
  });

  expect(pretty(container?.innerHTML)).toMatchSnapshot();
});
