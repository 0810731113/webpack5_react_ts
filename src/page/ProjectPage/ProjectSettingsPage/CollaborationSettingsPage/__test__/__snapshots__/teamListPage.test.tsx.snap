// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test team list page 1`] = `
"<div>
  <div class=\\"split-toolbar\\"></div>
  <div class=\\"ant-table-wrapper\\">
    <div class=\\"ant-spin-nested-loading\\">
      <div class=\\"ant-spin-container\\">
        <div class=\\"ant-table\\">
          <div class=\\"ant-table-container\\">
            <div class=\\"ant-table-content\\">
              <table style=\\"table-layout: auto;\\">
                <colgroup></colgroup>
                <thead class=\\"ant-table-thead\\">
                  <tr>
                    <th class=\\"ant-table-cell\\">团队名称</th>
                    <th class=\\"ant-table-cell\\">成员</th>
                    <th class=\\"ant-table-cell\\">操作</th>
                  </tr>
                </thead>
                <tbody class=\\"ant-table-tbody\\">
                  <tr data-row-key=\\"f058f8e7-b946-4e3f-8a8a-956146f881b8\\" class=\\"ant-table-row ant-table-row-level-0\\">
                    <td class=\\"ant-table-cell\\"><a href=\\"//f058f8e7-b946-4e3f-8a8a-956146f881b8\\">机电团队</a></td>
                    <td class=\\"ant-table-cell\\"><span class=\\"ant-tag\\">诸大大的工作室</span><span class=\\"ant-tag\\">老赵</span><span class=\\"ant-tag\\">大梦</span><span class=\\"ant-tag\\">豆豆</span></td>
                    <td class=\\"ant-table-cell\\">
                      <div class=\\"ant-space ant-space-horizontal ant-space-align-center\\">
                        <div class=\\"ant-space-item\\" style=\\"margin-right: 8px;\\"></div>
                        <div class=\\"ant-space-item\\"></div>
                      </div>
                    </td>
                  </tr>
                  <tr data-row-key=\\"6d52f9de-2c67-476c-94f3-c143b83d6ccf\\" class=\\"ant-table-row ant-table-row-level-0\\">
                    <td class=\\"ant-table-cell\\"><a href=\\"//6d52f9de-2c67-476c-94f3-c143b83d6ccf\\">结构团队</a></td>
                    <td class=\\"ant-table-cell\\"><span class=\\"ant-tag\\">诸大大的工作室</span><span class=\\"ant-tag\\">老赵</span><span class=\\"ant-tag\\">大梦</span><span class=\\"ant-tag\\">豆豆</span></td>
                    <td class=\\"ant-table-cell\\">
                      <div class=\\"ant-space ant-space-horizontal ant-space-align-center\\">
                        <div class=\\"ant-space-item\\" style=\\"margin-right: 8px;\\"></div>
                        <div class=\\"ant-space-item\\"></div>
                      </div>
                    </td>
                  </tr>
                  <tr data-row-key=\\"c98c4230-f2c9-4af6-9e5e-7abcd2c39141\\" class=\\"ant-table-row ant-table-row-level-0\\">
                    <td class=\\"ant-table-cell\\"><a href=\\"//c98c4230-f2c9-4af6-9e5e-7abcd2c39141\\">建筑团队</a></td>
                    <td class=\\"ant-table-cell\\"><span class=\\"ant-tag\\">诸大大的工作室</span><span class=\\"ant-tag\\">老赵</span><span class=\\"ant-tag\\">大梦</span><span class=\\"ant-tag\\">豆豆</span></td>
                    <td class=\\"ant-table-cell\\">
                      <div class=\\"ant-space ant-space-horizontal ant-space-align-center\\">
                        <div class=\\"ant-space-item\\" style=\\"margin-right: 8px;\\"></div>
                        <div class=\\"ant-space-item\\"></div>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
